### PR TITLE
SDK: Use Biome

### PR DESCRIPTION
**Pull request by devlyn37**
## Description

The main crossmint service repository uses `biome` for linting & formatting. Not doing so within the sdk repo has caused a bunch of friction with each developer that joins Crossmint, it's also silly to have a different set of rules across the repos!

This PR:
- Adds Biome, with config file copied over (only changing ignored files) from the main repo
- Applies biome fixes (mostly adding `type` to and removing imports)
- Deletes a bunch of prettier and eslint stuff!

## Test plan

TS compiles, projects build & existing tests pass.

## Package updates

N/A

### CODE REVIEW

File: biome.json
Comment (devlyn37) [inline biome.json:1]: copied over from crossbit-main with adjustments to what we ignore

File: package.json [hunk 1]
```diff
@@ -14,42 +14,30 @@
...
-        "eslint-config-prettier": "9.1.0",
-        "eslint-plugin-prettier": "4.2.1",
-        "eslint-plugin-unused-imports": "2.0.0",
         "gh-pages": "6.0.0",
         "husky": "9.0.11",
         "jest": "29.6.4",
         "jest-environment-jsdom": "29.6.4",
         "lerna": "6.6.2",
         "lint-staged": "15.2.7",
-        "prettier": "2.8.8",
```
Comment (devlyn37) [inline package.json:52]: best feeling

Comment (jmderby) [reply package.json:52]: 🔥

**Review by jmderby — APPROVED** [review]
Tested this in my local, had some minor friction with my existing formatting tools but resolved relatively fast and biome works well. Discussed in person with some super minor changes.

**Pull request merged**